geo/geomfn: fix Node if the LineString is the same point

otan opened the pull request:
Release note (bug fix): Fix a bug where ST_Node on a LineString with the
same repeated points results in an error.

Resolves #65682

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65686)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm!

otan committed:
geo/geomfn: fix Node if the LineString is the same point

Release note (bug fix): Fix a bug where ST_Node on a LineString with the
same repeated points results in an error.

otan commented:
thanks

bors r=rafiss
